src/utils/mod.rs

```diff
@@ -17,3 +38,114 @@ impl SwallowResult for Result<(), Error> {
         }
     }
 }
+
+/// Well-known information about the blockhain network
+///
+/// Some of the logic in Oura depends on particular characteristic of the
+/// network that it's consuming from. For example: time calculation and bech32
+/// encoding. This struct groups all of these blockchain network specific
+/// values.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct ChainWellKnownInfo {
+    pub shelley_slot_length: u32,
+    pub shelley_known_slot: u64,
+    pub shelley_known_hash: String,
+    pub shelley_known_time: u64,
+    pub address_hrp: String,
+}
+
+impl ChainWellKnownInfo {
+    /// Hardcoded values for mainnet
+    pub fn mainnet() -> Self {
+        ChainWellKnownInfo {
+            shelley_slot_length: 1,
+            shelley_known_slot: 4492800,
+            shelley_known_hash: "aa83acbf5904c0edfe4d79b3689d3d00fcfc553cf360fd2229b98d464c28e9de"
+                .to_string(),
+            shelley_known_time: 1596059091,
+            address_hrp: "addr".to_string(),
+        }
+    }
+
+    /// Hardcoded values for testnet
+    pub fn testnet() -> Self {
+        ChainWellKnownInfo {
+            shelley_slot_length: 1,
+            shelley_known_slot: 1598400,
+            shelley_known_hash: "02b1c561715da9e540411123a6135ee319b02f60b9a11a603d3305556c04329f"
+                .to_string(),
+            shelley_known_time: 1595967616,
+            address_hrp: "addr_test".to_string(),
+        }
+    }
+
+    /// Uses the value of the magic to return either mainnet or testnet
+    /// hardcoded values.
+    pub fn try_from_magic(magic: u64) -> Result<ChainWellKnownInfo, Error> {
+        match magic {
+            MAINNET_MAGIC => Ok(Self::mainnet()),
+            TESTNET_MAGIC => Ok(Self::testnet()),
+            _ => Err("can't infer well-known chain infro from specified magic".into()),
+        }
+    }
+}
+
+impl Default for ChainWellKnownInfo {
+    fn default() -> Self {
+        Self::mainnet()
+    }
+}
+
+/// Entry point for all shared utilities
+pub struct Utils {
+    pub(crate) well_known: ChainWellKnownInfo,
+    pub(crate) time: Option<NaiveTime>,
+    pub(crate) bech32: Bech32Provider,
+}
+
+impl Utils {
+    pub fn new(well_known: ChainWellKnownInfo) -> Self {
+        Self {
+            time: NaiveTime::new(TimeConfig::from_well_known(&well_known)).into(),
+            bech32: Bech32Provider::new(Bech32Config::from_well_known(&well_known)),
+            well_known,
+        }
+    }
+}
```

Comment:
I'm not entirely sure if `Utils` is the best name for the whole thing but I don't want to interfere with terminology.